src/dotty/tools/dotc/repl/CombinedClassLoader.scala

```diff
@@ -0,0 +1,37 @@
+package dotty.tools
+package dotc
+package repl
+
+import scala.util.Try
+
+class CombinedClassLoader(loaders: List[ClassLoader]) extends ClassLoader {
```

Comment:
In general, I would advise against making a custom subclass of ClassLoader because it can very easily not do what you think it should do, and it's hard to debug, but I don't know enough about the situation here to say for sure.